ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -628,7 +642,12 @@ public void onClose(
             ));
       }
 
-      StreamsErrorCollector.notifyApplicationClose(applicationId);
+      final Object o =
+          query.getStreamsProperties()
+              .get(KsqlConfig.KSQL_INTERNAL_STREAMS_ERROR_COLLECTOR_CONFIG);
+      if (o instanceof StreamsErrorCollector) {
+        ((StreamsErrorCollector) o).cleanup();
+      }
```

Comment:
This is a bit mysterious, but since we are no longer relying on a static map of application ids to find the error collector, we need to get the reference to the specific collector we passed in to this Streams app. The clean thing to do would have been to add the error collector to the QueryMetadata, but I was staring to get pretty concerned about increasing the scope of changes in this PR.

Instead, we can just get the instance out of the config map the exact same way that Streams does to configure the deserialization exception handler.